magento/magento2
# Fix - Newrelic GraphQl transaction names not working

### Description (*)
Add transaction names to all new relic graphql transactions.

So many other transactions are properly named without having the `newrelicreporting/general/enable=1` flag set. 
- cron jobs             ( handled by [magento php level](https://github.com/magento/magento2/blob/7c6b6365a3c099509d6f6e6c306cb1821910aab0/app/code/Magento/NewRelicReporting/Plugin/StatPlugin.php#L62-L73), so long as the php extension is enabled )
- CLI commands   ( handled by [magento php level](https://github.com/magento/magento2/blob/7c6b6365a3c099509d6f6e6c306cb1821910aab0/app/code/Magento/NewRelicReporting/Plugin/CommandPlugin.php#L58-L60), so long as the php extension is enabled )
- controllers          ( [handled by new relic agent](https://docs.newrelic.com/docs/apm/agents/php-agent/frameworks-libraries/magento-specific-functionality/) )
- rest API               ( [handled by new relic agent](https://docs.newrelic.com/docs/apm/agents/php-agent/frameworks-libraries/magento-specific-functionality/))

Currently graphql works like this
- graphql   ( handled by magento php level, so long as the php extension is enabled, and the  magento extension is enabled with `newrelicreporting/general/enable=1` )


Having to enable this new relic reporting module to get GraphQL transaction names seems like a bit of an oversight. It is perfectly viable to have this setup

1.  install new relic in the `php.ini` level, new relic will auto detect that it is magento2
```ini
; Setting: newrelic.framework
; Type   : string
; Scope  : per-directory
; Default: empty (auto-detect framework)
; Info   : Disables automatic framework detection, telling the agent to
;          attempt to name transactions according to the specified framework.
;          Specifying "no_framework" will disable framework-related transaction
;          naming entirely. Please let us know at support.newrelic.com if you
;          encounter a failure with framework autodetection.
;
;          Must be one of the following values:
;          cakephp, codeigniter, drupal, drupal8, joomla, kohana, laravel,
;          magento, magento2, mediawiki, silex, slim, symfony1, symfony2,
;          wordpress, yii, zend, zend2, no_framework
;
;          Note that "drupal" covers only Drupal 6 and 7.
;
;newrelic.framework = ""
```

2. Do not enable the magento extension, set `newrelicreporting/general/enable=0`
4. Expect to see transaction names for graphql

What we actually see is 

```
/FrontController/Magento\GraphQl\Controller\GraphQl\Interceptor
```

### Related Pull Requests
- https://github.com/magento/magento2/issues/36112
- https://github.com/magento/magento2/issues/30915

### Manual testing scenarios (*)
1. Configure new relic in your php.ini
2. Make a graphql request like `https://example.com/graphql?luker=3&query=%257Bproducts(filter%253A%257Bsku%253A%257%5B%25E2%2580%25A6%5DA%2522nonsuchsku%2522%257D%257D)%257Bitems%257Bname%2520sku%257D%257D%257D`
6. See `Graphql-products` in your new relic transactions list

![Screenshot 2022-11-08 at 15 24 31](https://user-images.githubusercontent.com/600190/200606503-feb7567a-5087-49a9-8181-3f1423e913fb.png)

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#36874: Fix - Newrelic GraphQl transaction names not working
